From d9d99097a5b28f4204b3a15e82564e50f9e62f5a Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Sat, 3 Aug 2019 08:05:35 +0200
Subject: [PATCH] ICU-20751 Fix nios2 build

Nios2 supports double conversion, tested using qemu:

./main || echo "correct"
correct
uname -a
Linux buildroot 4.19.16 #4 Sat Aug 3 14:46:48 CEST 2019 nios2 GNU/Linux

Solves build error

In file included from double-conversion.h:42:0,
                 from number_decimalquantity.cpp:19:
double-conversion-utils.h:119:2: error: #error Target architecture was not detected as supported by Double-Conversion.
 #error Target architecture was not detected as supported by Double-Conversion.

detected by buildroot autobuilders:
http://autobuild.buildroot.net/results/91e/91eaec34708d91f8a05af189243be0b7cabce31b/

Patch sent upstream: https://github.com/unicode-org/icu/pull/725
Bug report: https://unicode-org.atlassian.net/browse/ICU-20751

[Bernd: Fixed path]
Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 source/i18n/double-conversion-utils.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/source/i18n/double-conversion-utils.h b/source/i18n/double-conversion-utils.h
index 1e44fcaa0e..0cf993929f 100644
--- a/source/i18n/double-conversion-utils.h
+++ b/source/i18n/double-conversion-utils.h
@@ -95,6 +95,7 @@ int main(int argc, char** argv) {
     defined(__ARMEL__) || defined(__avr32__) || defined(_M_ARM) || defined(_M_ARM64) || \
     defined(__hppa__) || defined(__ia64__) || \
     defined(__mips__) || \
+    defined(__nios2__) || \
     defined(__powerpc__) || defined(__ppc__) || defined(__ppc64__) || \
     defined(_POWER) || defined(_ARCH_PPC) || defined(_ARCH_PPC64) || \
     defined(__sparc__) || defined(__sparc) || defined(__s390__) || \
-- 
2.20.1

